MergerService/Utils/TaskUtils.cs

```diff
@@ -13,107 +14,154 @@ public class TaskUtils : ITaskUtils
+            this._maxAttempts = this._configuration.GetConfiguration<int>("TASK", "maxAttempts");
+
+            // Construct Json serializer settings
+            _jsonSerializerSettings = new JsonSerializerSettings();
+            _jsonSerializerSettings.Converters.Add(new StringEnumConverter());
+
+            _overseerUrl = this._configuration.GetConfiguration("TASK", "overseerUrl");
+            _jobManagerUrl = this._configuration.GetConfiguration("TASK", "jobManagerUrl");
             //TODO: add tracing
         }
 
         public MergeTask? GetTask(string jobType, string taskType)
         {
-            string baseUrl = this._configuration.GetConfiguration("TASK", "jobManagerUrl");
-            string relativeUri = $"tasks/{jobType}/{taskType}/startPending";
-            string url = new Uri(new Uri(baseUrl), relativeUri).ToString();
-            string taskData = this._httpClient.PostDataString(url, null, false);
+            // TODO: add heartbeat start method
 
-            if (taskData is null)
+            using (var dequeueActivity = this._activitySource.StartActivity("dequeue task"))
             {
-                return null;
-            }
+                string relativeUri = $"tasks/{jobType}/{taskType}/startPending";
+                string url = new Uri(new Uri(_jobManagerUrl), relativeUri).ToString();
+                string? taskData = this._httpClient.PostDataString(url, null);
 
-            try
-            {
-                var jsonSerializerSettings = new JsonSerializerSettings();
-                jsonSerializerSettings.Converters.Add(new StringEnumConverter());
-                return JsonConvert.DeserializeObject<MergeTask>(taskData, jsonSerializerSettings)!;
-            }
-            catch (Exception e)
-            {
-                this._logger.LogWarning(e, "Error serializing returned task");
-                return null;
+                if (taskData is null)
+                {
+                    return null;
+                }
+
+                try
+                {
+                    return JsonConvert.DeserializeObject<MergeTask>(taskData, this._jsonSerializerSettings)!;
+                }
+                catch (Exception e)
+                {
+                    this._logger.LogWarning(e, "Error deserializing returned task");
+                    return null;
+                }
             }
+
+            // TODO: add heartbeat stop method
         }
 
         public void NotifyOnCompletion(string jobId, string taskId)
         {
-            // Notify overseer on task completion
-            string baseUrl = this._configuration.GetConfiguration("TASK", "overseerUrl");
-            string relativeUri = $"tasks/{jobId}/{taskId}/completed";
-            string url = new Uri(new Uri(baseUrl), relativeUri).ToString();
-            _ = this._httpClient.PostDataString(url, null, false);
+            using (var activity = this._activitySource.StartActivity("notify overseer on task completion"))
```

Comment:
save activity into variable only if you there metadata should be add to the trace,
otherwise just use:
using (this._activitySource.StartActivity("dequeue task"))

Reply:
Removed.